### PR TITLE
Homework_1 is done

### PR DESCRIPTION
**Pull request by 4eKiryaonow**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -11,6 +11,95 @@
...
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven-compiler-plugin.version}</version>
+                <configuration>
+                    <source>${maven.compiler.source}</source>
+                    <target>${maven.compiler.target}</target>
+                </configuration>
```
Comment (shoggy) [inline pom.xml]: Please use only one way to configure language level: properties or direct plugin configuration
https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-source-and-target.html

File: pom.xml [hunk 2]
```diff
@@ -11,6 +11,95 @@
...
+            </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>${maven-surefire-plugin.version}</version>
+            </plugin>
```
Comment (shoggy) [inline pom.xml]: Looks like configuration is missed, because independently of selected profile maven runs all tests

File: src/test/java/com/epam/tc/hw1/TestDivide.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.epam.tc.hw1;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+import com.epam.tat.module4.Calculator;
+import org.testng.annotations.Test;
+
+public class TestDivide {
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw1/TestDivide.java]: as improvement you can add test that check division by zero

File: src/test/java/com/epam/tc/hw1/TestDivide.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    @Test(
+            dataProviderClass = DataProviderForCalculator.class,
+            dataProvider = "Data for divide"
+    )
+    void testDivideLong(long a, long b, long expected) {
+
+        Calculator calculator = new Calculator();
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw1/TestDivide.java]: Please move initialization if SUT (system under test) to dedicated method
This method is common for all your test classes, so extract this method to some parent class

File: src/test/java/com/epam/tc/hw1/TestDivide.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package com.epam.tc.hw1;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
```
Comment (shoggy) [inline src/test/java/com/epam/tc/hw1/TestDivide.java]: Please use different class
```suggestion
import static org.assertj.core.api.Assertions.assertThat;
```

File: src/test/resources/RunAllTestNG.xml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+
+<suite name="Run all tests">
+    <test name="All tests">
+        <classes>
+            <class name="com.epam.tc.hw1.TestSum"></class>
+            <class name="com.epam.tc.hw1.TestSub"></class>
+            <class name="com.epam.tc.hw1.TestDivide"></class>
+            <class name="com.epam.tc.hw1.TestMultiply"></class>
+        </classes>
```
Comment (shoggy) [inline src/test/resources/RunAllTestNG.xml]: Try to use `packages` instead of enumerating all classes

File: src/test/resources/RunMultiplyDivideTestNG.xml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+
+<suite name="Run multiply divide tests">
+    <test name="run multiply divide">
+        <classes>
+            <class name="com.epam.tc.hw1.TestDivide"></class>
+            <class name="com.epam.tc.hw1.TestMultiply"></class>
+        </classes>
```
Comment (shoggy) [inline src/test/resources/RunMultiplyDivideTestNG.xml]: Try to use `@Test(groups= {})` and corresponding TestNg `groups` selector

**Review by shoggy — CHANGES_REQUESTED** [review]
## Test structure requirements:
- [x] `.gitignore` includes `target, .iml, .idea, ...`
- [x] Maven project
- [x] POM xml has following sections:
  - [x] artifact attributes
   - `artifactId` should be in form `firstname_lastname` as described in the presentation from introduction (Dmitrii Lebedev)
  - [x] properties
  - [x] dependencies
  - [x] plugins
  - [x] profiles
- [x] `calculator-1.0.jar` is added as maven dependency
- [x] TestNG is used as a testing framework
- [x] For test maven build maven-surefire-plugin is used
- [x] Code conforms to the Java Code Convention
- [x] `maven-checkstyle-plugin` is configured to use specific rules and is run during  `validate` phase

## File placement requirements:
- [x] Classes should be placed into `src/test/java/com/epam/tc/hw1` package and sub-packages
- [x] Homework should be developed in new Java classes
- [x] Test methods' names should reflect checking functionality

## Tests requirements:
- [x] Checkstyle plugin should not produce errors
- [x] You should test several Calculator operations (at least 4)
- [x] Test for each operation should be placed in separate classes
- [x] Create a test suite, which could run all tests
- [x] Split test for the two groups and create a suite for each group:
  - [x] Tests for add and subtract operations
  - [x] Tests for multiply and divide operations
- [x] All test data should be provided to test through the DataProvider
- [x] Profiles
  - surefire plugin configuration is missed
  - [x] Create 3 maven profiles which allow choosing test suite
  - [x] by default should work profile which runs add and subtract operations tests
- [x] Test must be run using maven
- [x] AssertJ is used instead of basic TestNg asserts

**Review by shoggy — APPROVED** [review]
Looks good
In next PRs try to reduce empty lines usage
